libs/execution_context/ebpf_maps.c

```diff
@@ -394,6 +430,17 @@ typedef struct _ebpf_map_metadata_table
         _In_ const uint8_t* previous_key,
         _Out_ uint8_t* next_key,
         _Inout_opt_ uint8_t** next_value);
+    ebpf_result_t (*query_buffer)(
+        _In_ const ebpf_core_map_t* map, uint64_t index, _Outptr_ uint8_t** data, _Out_ uint64_t* consumer_offset);
+    ebpf_result_t (*return_buffer)(_In_ const ebpf_core_map_t* map, uint64_t index, uint64_t consumer_offset);
+    ebpf_result_t (*write_data)(_Inout_ ebpf_core_map_t* map, uint64_t flags, _In_ uint8_t* data, size_t data_size);
+    ebpf_result_t (*async_query)(
+        _In_ const ebpf_core_map_t* map,
+        uint64_t index,
+        _Inout_ ebpf_map_async_query_result_t* async_query_result,
+        _Inout_ void* async_context);
+    void (*query)(
```

Comment:
```suggestion
    void (*query_ring_buffer)(
```

`*query` sounds too generic.